daemon/main.go

```diff
@@ -160,10 +160,19 @@ func getKernelVersion() (*go_version.Version, error) {
 		log.WithError(err).Fatal("kernel version: NOT OK")
 	}
 	verStrs := strings.Split(string(verOut), ".")
-	if len(verStrs) < 2 {
+	if len(verStrs) < 3 {
 		return nil, fmt.Errorf("unable to get kernel version from %q", string(verOut))
 	}
-	return go_version.NewVersion(strings.Join(verStrs[:2], "."))
+	// We are assuming the kernel version will be something as:
+	// 4.9.17-040917-generic
+
+	// If verStrs is []string{ "4", "9", "17-040917-generic" }
+	// then we need to split the patch string to get a number.
```

Comment:
In my case it looks like:

`$ uname -a
Linux localhost.localdomain 4.14.0-rc7+ #187 SMP Thu Nov 23 16:26:44 CET 2017 x86_64 x86_64 x86_64 GNU/Linux`

Couple of more possible examples:
```
4.10.0
4.12.0+
4.12.8
4.14.0-rc7+
```
Would they get detected as well?

Reply:
I've added unit tests
```
                {"4.10.0", mustHaveVersion("4.10.0")},
		{"4.10", mustHaveVersion("4.10.0")},
		{"4.12.0+", mustHaveVersion("4.12.0")},
		{"4.12.8", mustHaveVersion("4.12.8")},
		{"4.14.0-rc7+", mustHaveVersion("4.14.0")},
		{"4.9.17-040917-generic", mustHaveVersion("4.9.17")},
		{"4.9.generic", mustHaveVersion("4.9.0")},
```